Source/WebKit/UIProcess/API/glib/WebKitDownload.cpp

```diff
@@ -549,7 +549,7 @@ void webkit_download_cancel(WebKitDownload* download)
 
     download->priv->isCancelled = true;
     download->priv->download->cancel([download = Ref { *download->priv->download }] (auto*) {
-        download->client().legacyDidCancel(download.get());
+        download->client().didFinish(download.get());
```

Comment:
IIUC using `didFinish()` will consider the download successful, but a cancelled download should be considered as failed. Shouldn't this call `didFail()` instead?

Reply:
`webkitDownloadFinished` already handles the cancellation case, it's covered by existing unit tests.